collector/netframework_clr.go

```diff
@@ -0,0 +1,1218 @@
+	TotalNumberofContentions         *prometheus.Desc
+
+	// .NET CLR Memory metrics
+	AllocatedBytes            *prometheus.Desc
+	FinalizationSurvivors     *prometheus.Desc
+	HeapSize                  *prometheus.Desc
+	PromotedBytes             *prometheus.Desc
+	NumberGCHandles           *prometheus.Desc
+	NumberCollections         *prometheus.Desc
+	NumberInducedGC           *prometheus.Desc
+	NumberofPinnedObjects     *prometheus.Desc
+	NumberofSinkBlocksinuse   *prometheus.Desc
+	NumberTotalCommittedBytes *prometheus.Desc
+	NumberTotalreservedBytes  *prometheus.Desc
+	TimeinGC                  *prometheus.Desc
+
+	// .NET CLR Remoting metrics
+	Channels                  *prometheus.Desc
+	ContextBoundClassesLoaded *prometheus.Desc
+	ContextBoundObjects       *prometheus.Desc
+	ContextProxies            *prometheus.Desc
+	Contexts                  *prometheus.Desc
+	TotalRemoteCalls          *prometheus.Desc
+
+	// .NET CLR Security metrics
+	NumberLinkTimeChecks *prometheus.Desc
+	TimeinRTchecks       *prometheus.Desc
+	StackWalkDepth       *prometheus.Desc
+	TotalRuntimeChecks   *prometheus.Desc
+
+	// Process whitelist and blacklist regexp
+	processWhitelistPattern *regexp.Regexp
+	processBlacklistPattern *regexp.Regexp
+
+	netclrCollectors netclrCollectorsMap
+}
+
+func newNETCLRCollector() (Collector, error) {
+	const subsystem = "netframework_clr"
+
+	enabled := netclrExpandEnabledCollectors(*netclrEnabledCollectors)
+	perfCounters := make([]string, len(enabled))
+	for _, c := range enabled {
+		perfCounters = append(perfCounters, netclrGetPerfObjectName(c))
+	}
+	addPerfCounterDependencies(subsystem, perfCounters)
+
+	const exceptionsSubsystem = subsystem + "exceptions"
+	const interopSubsystem = subsystem + "interop"
+	const jitSubsystem = subsystem + "jit"
+	const loadingSubsystem = subsystem + "loading"
+	const locksandthreadsSubsystem = subsystem + "locksandthreads"
+	const memorySubsystem = subsystem + "memory"
+	const remotingSubsystem = subsystem + "remoting"
+	const securitySubsystem = subsystem + "security"
+	netclrCollector := &netclrCollector{
+		// meta
+		netclrScrapeDurationDesc: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "collector_duration_seconds"),
+			"windows_exporter: Duration of a netframework_clr child collection.",
+			[]string{"collector"},
+			nil,
+		),
+		netclrScrapeSuccessDesc: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "collector_success"),
+			"windows_exporter: Whether a netframework_clr child collector was successful.",
+			[]string{"collector"},
+			nil,
+		),
+
+		// .NET CLR Exceptions metrics
+		NumberofExcepsThrown: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, exceptionsSubsystem, "exceptions_thrown_total"),
+			"Displays the total number of exceptions thrown since the application started. This includes both .NET exceptions and unmanaged exceptions that are converted into .NET exceptions.",
+			[]string{"process"},
```

Comment:
This was changed to `pid`, right?

Reply:
Good spot, that was my mistake forgetting to set the correct flag name. Thanks for that :+1: